### ISSUE TITLE
To/from jotdown events

### ISSUE THREAD

**Issue by clbarnes** [clbarnes/djot_ast#1]
See https://github.com/hellux/jotdown/issues/17

Building an AST from an iterator of jotdown events would give us parsing; converting to a stream of jotdown events would give us rendering, thereby allowing a full rusty roundtrip. Note that until then, you can always use the typescript impl to emit JSON and then render it, using this library just to operate on the JSON in the middle.

**Comment by not-my-profile** [comment]
I think this should ideally be implemented in Jotdown and not here. (It's nicer for a parser to depend on an AST crate than an AST crate to depend on a parser).

Furthermore I think when writing this crate you made a bit of an oversight in making all the block and inline types unconstructable for API users because they all have private fields ... (mostly the meta field ... in same cases like `BulletList` all fields are private). For an AST library I don't see any reason for any fields to be private.

**Comment by clbarnes** [comment]
Yes it's a pretty rough draft and those kinds of issues will spring up if/when it's actually used! Thanks for noting it.

I was thinking the to/from would be behind a feature anyway.
